pkg/proxy/proxy/transparent.go

```diff
@@ -101,8 +98,8 @@ type TransparentProxyService struct {
 // NewTransparentProxyService creates a new TransparentProxyService.
 func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode string) TransparentProxyService {
 	m := new(sync.Mutex)
-	pm := StringToProxyMode(mode)
-	if pm == UnkownMode {
+	pm := ProxyMode(mode)
```

Comment:
since this is part of the public pkg interface, might it make sense to change the type of the param?